packages/web/utils/metamask.ts

```diff
@@ -0,0 +1,94 @@
+import { NETWORK_INFO } from 'utils/networks';
+
+export const switchChainOnMetaMask = async (
+  chainId: string,
+): Promise<boolean> => {
+  if (!NETWORK_INFO[chainId]) {
+    // eslint-disable-next-line no-console
+    console.error(`No network configuration found for chainId ${chainId}`);
+    return false;
+  }
+
+  const { name, symbol, explorer, rpc } = NETWORK_INFO[chainId];
+
+  if (!(name && symbol && rpc && explorer && window.ethereum?.isMetaMask)) {
+    // eslint-disable-next-line no-console
+    console.error('Switching chain is only supported in Metamask');
+    return false;
+  }
+
+  try {
+    await window.ethereum.request({
+      method: 'wallet_switchEthereumChain',
+      params: [
+        {
+          chainId,
+        },
+      ],
+    });
+    return true;
+  } catch (switchError) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if ((switchError as { code?: number }).code === 4902) {
+      try {
+        await window.ethereum.request({
+          method: 'wallet_addEthereumChain',
```

Comment:
This will add the chain, [but not switch to it](//docs.metamask.io/guide/rpc-api.html#:~:text=the%20user%20may%20choose%20to%20switch%20to%20the%20chain%20once%20it%20has%20been%20added.).

You could call `switchChainOnMetaMask` recursively if this succeeds.

Reply:
I've tested it. And it doesn switch chains after adding it.